src/services/facilityService.ts

```diff
@@ -220,6 +220,8 @@ export const updateFacility = async (facilityId: string, fieldsToUpdate: Partial
         const dbDocument = await facilityRef.get()
         const dbFacilityToUpdate = dbDocument.data() as dbSchema.Facility
 
+        const originalHealthcareProfessionalIdsForFacility = dbFacilityToUpdate.healthcareProfessionalIds
```

Comment:
Before merging, double check that we don't need a deep copy here. If the references are updated the new list be as well

Reply:
The deep copy is happening within the function where this is being used. Would it be safer to be a deep copy in both places. But it is being passed and then deep copied within that function.

Reply:
I think that it is being handled well on line 294. So I think we are good! 

```
async function processHealthcareProfessionalRelationshipChanges(facilityId: string,
    healthcareProfessionalRelationshipChanges: gqlTypes.Relationship[],
    batch: WriteBatch,
    originalHealthcareProfessionalIds: string[] = [])
    : Promise<Result<string[]>> {
    // deep clone the array so we don't modify the original
    let updatedProfessionalIdsArray = [...originalHealthcareProfessionalIds]

    healthcareProfessionalRelationshipChanges.forEach(change => {
        switch (change.action) {
            case gqlTypes.RelationshipAction.Create:
                updatedProfessionalIdsArray.push(change.otherEntityId)
                break
            case gqlTypes.RelationshipAction.Delete:
                updatedProfessionalIdsArray = updatedProfessionalIdsArray.filter(id => id !== change.otherEntityId)
                break
            default:
                break
        }
    })
    ```

You even made a note of it!